### ISSUE TITLE
filesystem_check_changes for memories

### ISSUE THREAD

**Issue by farline99** [pulsejet/memories#939]
my photos/videos are synced via syncthing from phone to the server. in nextcloud i enabled the filesystem_check_changes option in config.php and the files app correctly displays new photos/videos immediately after syncthing transfer, but not memories. i don't want the "to run the occ files:scan and occ memories:index" commands regularly so that memories just sees my photos.
is it possible to make filesystem_check_changes for memories?

**Comment by pulsejet** [comment]
tl;dr no, this is unlikely to work

AFAIK, `filesystem_check_changes` validates the filecache every time you browse to a directory. Memories on the other hand _relies_ on the filecache to already exist, so that it can make very fast database queries. This means there are a few ways to make this work.

1. Memories must scan/validate your timeline root recursively for every request. This will be ridiculously slow.
2. Have a separate process watch changes to the filesystem and trigger Nextcloud hooks. Something like this should be implemented in the Nextcloud core and not a specialized app.
3. (most sensible) we could have an option to revalidate the filecache at the start of the scheduled cron job. But this will mean a 15-20 minute delay for files to show up in Memories.

In general workflow that needs `filesystem_check_changes` would likely not play nice since no hooks are triggered for file creation, modification etc. For best results, all filesystem updates _must_ happen through Nextcloud's WebDAV interface.

**Comment by farline99** [comment]
understood you, unfortunately there were problems with webdav, so I had to resort to syncthing. thanks for the answer!
